internal/cache/cache.go

```diff
@@ -73,6 +73,14 @@ func (c *Cache) Get(key string, fetch func() (interface{}, error)) (interface{},
 	return c.getAtTime(key, fetch, time.Now())
 }
 
+// Refresh returns the value for the given key, using fetch to fetch
+// the value and store the value in the cache refreshing any previous value.
+// If fetch returns an error, the returned error from Refresh will have
+// the same cause.
+func (c *Cache) Refresh(key string, fetch func() (interface{}, error)) (interface{}, error) {
```

Comment:
Rather than duplicating the whole signature of Get, how about just adding
a more specific Evict method (since we've already got EvictAll):

    // Evict removes the entry with the given key from the cache.
    func (c *Cache) Evict(key string)
        
This would make for simpler tests and code, I think.